### ISSUE TITLE
@nuxt/content 2.13.1 -> ERROR  [unhandledRejection] connect ECONNREFUSED 127.0.0.1:49363

### ISSUE THREAD

**Issue by XStarlink** [nuxt/content#2691]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.15.0`
- Nuxt Version:     `3.12.3`
- CLI Version:      `3.12.0`
- Nitro Version:    `2.9.7`
- Package Manager:  `npm@10.7.0`
- Builder:          `-`
- User Config:      `future`, `compatibilityDate`, `ssr`, `telemetry`, `css`, `typescript`, `build`, `vite`, `routeRules`, `nitro`, `extends`, `modules`, `app`, `site`, `ogImage`, `sitemap`, `robots`, `seoExperiments`, `schemaOrg`, `linkChecker`, `ui`, `colorMode`, `googleFonts`, `i18n`, `vueEmail`, `supabase`, `eslint`, `turnstile`, `devtools`, `runtimeConfig`
- Runtime Modules:  `@nuxt/eslint@0.3.13`, `@nuxt/scripts@0.5.1`, `@nuxt/content@2.13.1`, `@nuxt/image@1.7.0`, `@nuxt/ui@2.17.0`, `@nuxtjs/turnstile@0.9.0`, `@nuxtjs/i18n@8.3.1`, `@nuxtjs/seo@2.0.0-rc.11`, `@nuxtjs/supabase@1.3.2`, `@nuxtjs/google-fonts@3.2.0`, `@pinia/nuxt@0.5.1`, `@vueuse/nuxt@10.11.0`, `@formkit/auto-animate/nuxt@0.8.2`, `()`
- Build Modules:    `-`

### Reproduction

.

### Describe the bug

Hi, when I upgrade from @nuxt/content v2.13.0 to v2.13.1 and make any change in `/server`, I get this error in console:

```
 ERROR [unhandledRejection] connect ECONNREFUSED 127.0.0.1:49363 18:49:19
  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1606:16)
```

I can't even refresh the app page which loads indefinitely, and even restarting the dev server doesn't solve the problem. I have to delete the .nuxt folder and restart.

And when I go back to @nuxt/content v2.13.0 I don't have the problem anymore.

I researched the error, and came across this issue: https://github.com/nuxt/nuxt/issues/23832

It seems to be related to Vite and HMR when using plugins with addVitePlugin, and [comparing the code of version v2.13.0 with v2.13.1](https://github.com/nuxt/content/compare/v2.13.0...v2.13.1) we see that the use of Vite and plugins has been modified, so it may come from there.

I'll try to make a reproduction as soon as I have a bit more time.

**Comment by yurijmikhalevich** [comment]
Faced the same issue on macOS; switching to node v18 fixed it for me.

**Comment by prostohttp** [comment]
Same issues  on macOS; switching to node v18 didn't help me. I get this error, for example, when saving a mongoose schema or when changing files.

**Comment by XStarlink** [comment]
I've just done a local `npx nuxi@latest upgrade --force` with the latest version of content installed before (v2.13.1), and I don't seem to get the error anymore, it was probably a problem with dependencies that didn't update well.

Thank you very much @farnabaz for your advice (normally I always test this, sorry to have wasted your time...), maybe we can close the issue.

On the other hand now with the clean install (which I do with each new version of Nuxt), I get this from time to time in the dev server console, but it's not related to nuxt content sorry for the irrelevance:

```
 ERROR [unhandledRejection] write EPIPE 18:53:29

  at afterWriteDispatched (node:internal/stream_base_commons:161:15)
  at writeGeneric (node:internal/stream_base_commons:152:3)
  at Socket._writeGeneric (node:net:953:11)
  at Socket._write (node:net:965:8)
  at writeOrBuffer (node:internal/streams/writable:564:12)
  at _write (node:internal/streams/writable:493:10)
  at Writable.write (node:internal/streams/writable:502:10)
  at Socket.ondata (node:internal/streams/readable:1007:22)
  at Socket.emit (node:events:519:28)
  at addChunk (node:internal/streams/readable:559:12)
  ```

I'd already had it a few months ago, then it was gone, and now it's back.

**Comment by CptJJ** [comment]
> Same issues on macOS; switching to node v18 didn't help me. I get this error, for example, when saving a mongoose schema or when changing files. I also had "@nuxt/content": "^2.13.1" installed, changing the version to "@nuxt/content": "^2.13.0" did not solve the problem.
> 
> <img alt="Снимок экрана 2024-07-08 в 10 18 16" width="622" src="https://private-user-images.githubusercontent.com/33367694/346384142-4b7525fd-359c-4525-b043-77923f383b35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA1NDkwODAsIm5iZiI6MTcyMDU0ODc4MCwicGF0aCI6Ii8zMzM2NzY5NC8zNDYzODQxNDItNGI3NTI1ZmQtMzU5Yy00NTI1LWIwNDMtNzc5MjNmMzgzYjM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA5VDE4MTMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY3MzFhNTY4NzMxMGU4Mjk3MGJjNjYzMGU3MTUxNDAxZjE4MzU0MTdmMjQzOTJjYWFkZTBkODY4OTc2MzUyN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.j0AaO57r-zEnn7xIwfHLOvPw37kGjMJChAUHEI1Zx5U">

Same issue here started happening a few days ago and have been unable to chase it down.

Seems larger than nuxt/content? I get lots of vite errors when running pnpm install maybe its related

![CleanShot 2024-07-09 at 12 02 00@2x](https://github.com/nuxt/content/assets/5834717/0b1f0d5e-15e1-4bff-8d62-a1fb5bafa5b9)

**Comment by alexmercier25** [comment]
same issue here, also started happening a few days ago. 
```
✔ Nuxt Nitro server built in 3080 ms                                                                                                    nitro 3:44:56 PM

 ERROR  [unhandledRejection] connect ECONNREFUSED 127.0.0.1:65146                                                                              3:44:56 PM

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1607:16)


```

it happens everytime I make a change on the `server` folder.. Then I have to stop the server, remove the `.nuxt` folder and start the server again..

**Comment by CptJJ** [comment]
I fixed all the vite errors by install vite by itself via pnpm install vite... fixed HRM issue also

**Comment by narr07** [comment]
same issue

- Nuxt 3.12.3 
- Nitro 2.9.7
- @nuxt/content:  2.13.1
- @nuxt/ui: 2.17.0

``` 
ERROR  [unhandledRejection] connect ECONNREFUSED 127.0.0.1:56891              10:22:35 AM

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1606:16)

**Comment by prostohttp** [comment]
Guys, if you have [https://marketplace.visualstudio.com/items?itemName=Nuxtr.nuxtr-vscode](https://marketplace.visualstudio.com/items?itemName=Nuxtr.nuxtr-vscode) installed, then do not update any dependencies using it, after it I get an error again.
Here's the solution

**_npx nuxi@latest upgrade_**

We are waiting for the new version of the extension

**Comment by chrisjansky** [comment]
Not sure this will help anyone—I’ve been literally tearing my hair out these past 2 days over this inexplicable error and finally what seemed to have helped was reverting to `nuxt@3.12.1` and `@nuxt/content@2.13.0` (at least for the time being)

**Comment by cossssmin** [comment]
Updating `@nuxt/content` from `2.13.0` to `2.13.1` resulted in no content page being generated. Netlify built successfully so I only caught it by chance, browsing the site and not seeing the content sections :( Reverted to `2.13.0` in the meantime.

**Comment by farnabaz** [comment]
@cossssmin Checkout release notes of `2.13.1`. https://github.com/nuxt/content/releases/tag/v2.13.1

Due to some regression in Nuxt you need to add `/` to pre-render routes

**Comment by danielroe** [comment]
The regression @farnabaz refers to will be resolved in the next patch release of Nuxt.

**Comment by XStarlink** [comment]
Finally, I come back to what I said at the very beginning of this issue,

I really think that @nuxt/content@2.13.1 has something to do with the HMR error (which is probably linked to a problem on the Vite side), because for the past two weeks, as soon as I change a line on the /server side, I've had to delete the .nuxt and restart (which makes development extremely constraining, even though I love Nuxt so much...), because after each change I've had the error:

 ERROR [unhandledRejection] connect ECONNREFUSED 127.0.0.1:49363 18:49:19
  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1606:16)

Today I ran several tests, and by re-installing @nuxt/content@2.13.0 I no longer have the error at all! (What a relief!)
So if you get this error, try to go back to v2.13.0, and if ever tell us here if it solves your problem or not.

**Comment by farnabaz** [comment]
@XStarlink @wenxb This might be related to the new file watcher in dev mode, and should be fixed in https://github.com/nuxt/content/commit/ec7105ad2280624e07529bdedc822bb97344d3c4

You can check with [edge channel](https://content.nuxt.com/get-started/edge-channel)

**Comment by alexmercier25** [comment]
> @XStarlink @wenxb This might be related to the new file watcher in dev mode, and should be fixed in [ec7105a](https://github.com/nuxt/content/commit/ec7105ad2280624e07529bdedc822bb97344d3c4)
> 
> You can check with [edge channel](https://content.nuxt.com/get-started/edge-channel)

this worked for me! No more ECONNREFUSED error when saving server-side.

thank you!!

**Comment by farnabaz** [comment]
Thanks for confirming the fix @alexmercier25, I'll release a patch ASAP.

**Comment by farnabaz** [comment]
Fixed in [v2.13.2](https://github.com/nuxt/content/releases/tag/v2.13.2)

**Comment by XStarlink** [comment]
I confirm that v2.13.2 fixes my ECONNREFUSED problem.

Thank you very much for your work @farnabaz !!
